### PR TITLE
Chrysler non-ACC Fix - Only allow openpilot in ACC mode

### LINKED ISSUE #1278: ACC can by bypassed on Chrysler
ACC can by bypassed on Chrysler. Openpilot allows engagement when ACC is disabled.

How to reproduce or log data
Press the cruise on/off button instead of ACC on/off button. ACC will be disabled and the car will switch to non-adaptive cruise control. Press set to engage. Openpilot will control steering, but ACC will not function.

Expected behavior
Openpilot should not be allowed to be engaged without ACC.

Device/Version information:

Device: EON Gold
Version: 0.7.4
Car make/model: 2017 Chrysler Pacifica (Gas)

### PR DESCRIPTION
**Pull request by vanillagorillaa**
Fixes https://github.com/commaai/openpilot/issues/1278

Still needs to be tested and awaiting a route. Should probably also look for other states in the message.

### CODE REVIEW

File: selfdrive/car/chrysler/carstate.py [hunk 1]
```diff
@@ -49,6 +49,8 @@ def update(self, cp, cp_cam):
     ret.cruiseState.enabled = cp.vl["ACC_2"]['ACC_STATUS_2'] == 7  # ACC is green.
     ret.cruiseState.available = ret.cruiseState.enabled  # FIXME: for now same as enabled
     ret.cruiseState.speed = cp.vl["DASHBOARD"]['ACC_SPEED_CONFIG_KPH'] * CV.KPH_TO_MS
+    #CRUISE_STATE is a three bit msg, 0 is off, 1 and 2 are Non-ACC mode, 3 and 4 are ACC mode, find if there are other states too
+    ret.cruiseState.nonAdaptive = cp.vl["DASHBOARD"]['CRUISE_STATE'] in [1, 2]
```
Comment (adeebshihadeh) [inline selfdrive/car/chrysler/carstate.py:53]: need to add this new signal to the list of can parser signals

File: selfdrive/car/chrysler/carstate.py [hunk 2]
```diff
@@ -95,6 +95,7 @@ def get_can_parser(CP):
       ("COUNTER", "EPS_STATUS", -1),
       ("TRACTION_OFF", "TRACTION_BUTTON", 0),
       ("SEATBELT_DRIVER_UNLATCHED", "SEATBELT_STATUS", 0),
+      ("CRUISE_STATE", "DASHBOARD", 0),
```
Comment (adeebshihadeh) [inline selfdrive/car/chrysler/carstate.py:98]: move this next to the other dashboard signal

**Comment by vanillagorillaa** [comment]
522b7df1fa94de66|2020-12-28--21-37-14

**Comment by pd0wm** [comment]
Merged opendbc pr and updated on master. Please rebase

**Comment by pd0wm** [comment]
Still changes to opendbc

File: selfdrive/car/chrysler/carstate.py [hunk 3]
```diff
@@ -49,6 +49,8 @@ def update(self, cp, cp_cam):
     ret.cruiseState.enabled = cp.vl["ACC_2"]['ACC_STATUS_2'] == 7  # ACC is green.
     ret.cruiseState.available = ret.cruiseState.enabled  # FIXME: for now same as enabled
     ret.cruiseState.speed = cp.vl["DASHBOARD"]['ACC_SPEED_CONFIG_KPH'] * CV.KPH_TO_MS
+    #CRUISE_STATE is a three bit msg, 0 is off, 1 and 2 are Non-ACC mode, 3 and 4 are ACC mode, find if there are other states too
+    ret.cruiseState.nonAdaptive = cp.vl["DASHBOARD"]['CRUISE_STATE'] in [1,2]
```
Comment (adeebshihadeh) [inline selfdrive/car/chrysler/carstate.py:53]:
```suggestion
    # CRUISE_STATE is a three bit msg, 0 is off, 1 and 2 are Non-ACC mode, 3 and 4 are ACC mode, find if there are other states too
    ret.cruiseState.nonAdaptive = cp.vl["DASHBOARD"]['CRUISE_STATE'] in [1, 2]
```

**Pull request merged**